packages/server/src/scripts/parse_and_cache_data.js

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env node
+/* eslint-disable */
+
+require("dotenv").config();
+
+const { loadRecordsForUpload } = require('../arpa_reporter/services/records')
+const _ = require("lodash");
+
+const inquirer = require("inquirer");
+
+// This is so named to keep from accidentally typing "knex" out of habit and thus having some
+// queries inadvertently outside the transaction.
+const knexWithoutTransaction = require("../db/connection");
+const { getUpload, cacheRecords } = require("../arpa_reporter/db/uploads");
+
+async function main() {
+    if (!process.env.POSTGRES_URL) {
+        console.error("must specify POSTGRES_URL env variable");
+        return;
+    }
+    console.log(
+        "Parsing and caching upload data with postgres db: ",
+        process.env.POSTGRES_URL
+    );
+
+    const { uploadId } = await inquirer.prompt([
+        {
+            type: "input",
+            name: "uploadId",
+            message: "What is the uuid of the upload you want to parse and cache:",
+        },
+    ])
+
+    console.log("Parsing and caching upload with id:", uploadId);
+
+
+    const upload = await getUpload(uploadId)
+    const records = await loadRecordsForUpload(upload)
+
+    await cacheRecords(upload, records)
+
+    // Now that the data has been written, check that it matches what we parsed from excel
+    const updatedUpload = await getUpload(uploadId)
+
+    if (_.isEqual(records, updatedUpload.parsed_data)) {
+        console.log("Cached data is equivalent to parsed data")
+    } else {
+        diff = compare(records, updatedUpload.parsed_data)
+        console.log("ERROR: Cached and parsed data are not equivalent")
+        console.log(diff)
+    }
+
+}
+
+// This method is copied verbatim from https://stackoverflow.com/a/41431685/21046021
+// and allows us to see every json path that differs between two objects
+var compare = function (a, b) {
```

Comment:
This works, but I wonder about using a library like [deep diff](https://www.npmjs.com/package/deep-diff) ? I like the additional info this provides as well. It is yet another library, but it seems to handle this particular use case well and seems to have pretty good adoption overall (1mm downloads per week!)

Reply:
I would also prefer to declare a dependency over copy-pasting code.  👍

Reply:
sounds good, I'll update this!